### PR TITLE
FIX: interpolate_missing type check against BaseEpochs

### PR DESCRIPTION
**Pull request by nfoti**
Addresses #50

### CODE REVIEW

**Comment by jasmainak** [comment]
LGTM @dengemann okay with you?

**Comment by dengemann** [comment]
@nfoti can you comment on this? I think we need(ed) the check for EpochsArray at some point.

**Comment by nfoti** [comment]
EpochsArray derives from BaseEpochs, so should be fine.

**Comment by jasmainak** [comment]
yeah, you needed it before because `BaseEpochs` was private, but now it's not the case ...

**Comment by nfoti** [comment]
@dengemann @jasmainak are we good to merge?

**Comment by dengemann** [comment]
I am waiting for you to explaine me why this is necessary :)

**Comment by nfoti** [comment]
I thought it was pretty clearly spelled out in #50 and this thread, but to explain it again:

The `interpolate_missing` function performs a type check for `Epochs` and `EpochsArray` objects. However, loading epochs from files return objects of type `EpochsFIF` which fail the type check. However, all of these objects are of type `BaseEpochs` which is a public class now, so we can just type check against `BaseEpochs` rather than all three types.

**Comment by agramfort** [comment]
LGTM

+1 for MRG

**Comment by dengemann** [comment]
Don't merge please! I will take a look later.

File: hcp/preprocessing.py [hunk 1]
```diff
@@ -214,7 +214,7 @@ def interpolate_missing(inst, subject, data_type, hcp_path,
     is_raw = isinstance(inst, (mne.io.Raw,
                                mne.io.RawArray,
                                mne.io.bti.bti.RawBTi))
-    is_epochs = isinstance(inst, (mne.Epochs, mne.EpochsArray))
+    is_epochs = isinstance(inst, mne.BaseEpochs)
```
Comment (jasmainak) [inline hcp/preprocessing.py]: actually, to make sure that you are compliant with older versions of MNE, you should do something like this: https://github.com/autoreject/autoreject/blob/master/autoreject/utils.py#L61-L64

Comment (nfoti) [inline hcp/preprocessing.py]: that's a good catch.

**Comment by nfoti** [comment]
I'm leaving this simple fix to @dengemann and @jasmainak at this point.

**Comment by dengemann** [comment]
Thanks @nfoti merged by rebase.

**Pull request closed without merging**